src/Poseidon/EntityTypes.hs

```diff
@@ -0,0 +1,165 @@
+{-# LANGUAGE DeriveGeneric     #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module Poseidon.EntityTypes (
+    IndividualInfo (..), getIndName, PackageInfo (..), GroupInfo (..), ExtendedIndividualInfo (..),
+    renderNameWithVersion,
+    HasNameAndVersion (..),
+    PacNameAndVersion(..), PoseidonIndividual(..), makePacNameAndVersion) where
+
+import           Data.Aeson   (FromJSON (..), KeyValue ((.=)), ToJSON (..),
+                               object, withObject, (.:))
+import           Data.Time    (Day)
+import           Data.Version (Version, showVersion)
+import           GHC.Generics (Generic)
+
+-- Entity definiting data types
+
+-- | A datatype to represent a package, a group or an individual
+data PoseidonIndividual =
+      SimpleInd String
+    | SpecificInd IndividualInfo
+    deriving (Eq, Ord)
+
+getIndName :: PoseidonIndividual -> String
+getIndName (SimpleInd n)                        = n
+getIndName (SpecificInd (IndividualInfo n _ _)) = n
+
+instance Show PoseidonIndividual where
+    show (SimpleInd   i) = "<" ++ i ++ ">"
+    show (SpecificInd i) = show i
+
+data IndividualInfo = IndividualInfo
+    { indInfoName   :: String
+    , indInfoGroups :: [String]
+    , indInfoPac    :: PacNameAndVersion
+    } deriving (Ord, Generic)
+
+instance Eq IndividualInfo where
```

Comment:
This overloading of the `Eq` class makes me quite uncomfortable. I wonder whether this will bite is in the back some time. Say we want to check equality for testing purposes or so... I think I would prefer not doing this and instead use a custom function like `compareIndInfo` or something.